ILTransform/ILRewriter.cs

```diff
@@ -97,6 +97,24 @@ private void RewriteFile(string source)
                 int lineIndex = _testProject.LastMainMethodDefLine;
                 if (lineIndex >= 0)
                 {
+                    if (_settings.CollapseMainSign
+                        && (_testProject.FirstMainMethodDefLine != -1)
+                        && (_testProject.FirstMainMethodDefLine < _testProject.LastMainMethodDefLine))
+                    {
+                        int mainLine = _testProject.FirstMainMethodDefLine;
+                        string collapsedLine = lines[mainLine].TrimEnd();
+                        for(mainLine++; mainLine <= _testProject.LastMainMethodDefLine; mainLine++)
+                        {
+                            collapsedLine += " " + lines[mainLine].Trim();
+                        }
+                        // overwrite first main line and remove the rest
+                        lines[_testProject.FirstMainMethodDefLine] = collapsedLine;
+                        int mainSignLineLength = _testProject.LastMainMethodDefLine - _testProject.FirstMainMethodDefLine;
+                        lines.RemoveRange(_testProject.FirstMainMethodDefLine+1, mainSignLineLength);
```

Comment:
Is this removing an extra line?

Reply:
This always confuses me. So, the substruction would give the number of items as if it is [,). For example, if it is a main sign in two lines, `mainSignLineLength = 1` and we have to remove from `_testProject.FirstMainMethodDefLine+1` (the second line)